machine_test.go

```diff
@@ -1645,3 +1652,22 @@ func TestCreateSnapshot(t *testing.T) {
 		})
 	}
 }
+
+func testAttachBalloonDevice(ctx context.Context, t *testing.T, m *Machine){
```

Comment:
It is a bit strange to see this test is passing Machine's member to Machine itself. Does it have to be coming from m.Cfg?

Reply:
I'm also thinking to refactor it.